numpydoc/tests/test_validate.py

```diff
@@ -41,6 +41,33 @@ def test_get_validation_checks_validity(checks):
         _ = validate.get_validation_checks(checks)
 
 
+class _DummyList(list):
+    """Dummy list class to test validation."""
+
+
+def test_no_file():
+    """Test that validation can be done on functions made on the fly."""
+    # Just a smoke test for now, <list> will have a None filename
+    validate.validate("numpydoc.tests.test_validate._DummyList.clear")
+    # This does something like decorator.FunctionMaker.make
+    src = """\
+def func():
+    '''Do something.'''
+    return 1
+"""
+    evaldict = {}
+    exec(compile(src, "<string>", "single"), evaldict)
+    func = evaldict["func"]
+    func.__source__ = src
+    func.__module__ = "numpydoc.tests.test_validate"
+    assert func() == 1
+    # This should get past the comment reading at least. A properly
+    # wrapped function *would* have source code, too. We could add such
+    # a test later
+    with pytest.raises(OSError, match="could not get source code"):
```

Comment:
Bringing it back to my [earlier question](https://github.com/numpy/numpydoc/pull/476#discussion_r1297785873), this would mean that validation wasn't run in these cases because it would have failed during the ast parsing [here](https://github.com/numpy/numpydoc/blob/main/numpydoc/validate.py#L485)  (4 year old code), and therefore not returned any findings. In this case, I wonder if it would make more sense to be explicit why this happens in an error message or support reporting on a subset of the checks that don't require this.

Reply:
Maybe but I want to stress this is really a corner case I only hit while trying to find the *real motivating* case above. So actually I think it's better for now if I just remove this so we aren't confused by it...